Makefile

```diff
@@ -105,7 +106,14 @@ test.ci: generate
 .PHONY: e2e
 e2e: $(call GETENVOY_OUT_PATH,$(GOOS),$(GOARCH))
 	docker-compose up -d
-	go test $(E2E_OPTS) $(E2E_EXTRA_OPTS) $(E2E_PKG_LIST)
+	$(RUN_E2E)
+
+# Runs e2e for all extensions languages in pkg/extension/workspace/config/extension/helpers.go
+.PHONY: e2e.all
+e2e.all: $(call GETENVOY_OUT_PATH,$(GOOS),$(GOARCH))
+	docker-compose up -d
+	E2E_EXTENSION_LANGUAGE=rust $(RUN_E2E)
+	E2E_EXTENSION_LANGUAGE=tinygo $(RUN_E2E)
```

Comment:
not sure if the test results would be cached from the previous run for Rust here..

Reply:
I think the cache concern is addressed now, since all matrices materialize in go even if externally seeded?

Reply:
PS on my laptop `-test.failfast` seems to invalidate cache, this is set by default in make so that CI runs don't proceed forever.. we can remove that with a comment.

Reply:
but to answer your question, yeah it seems that go caches the name, not the matrix. So, the above (which no longer exists) wouldn't have cached properly!

Reply:
added notes about the cache behavior thanks, as I didn't know this! https://github.com/golang/go/issues/42544#issuecomment-726204341